core/trino-main/src/main/java/io/trino/operator/TopNProcessor.java

```diff
@@ -46,28 +50,35 @@ public TopNProcessor(
             List<Type> types,
             int n,
             List<Integer> sortChannels,
-            List<SortOrder> sortOrders, TypeOperators typeOperators)
+            List<SortOrder> sortOrders,
+            TypeOperators typeOperators,
+            Optional<DataSize> maxPartialMemory)
```

Comment:
TopNProcessor should probably not need to be aware of partials. Partials is an operator concept that should be handled externally to this class.

Reply:
Done